std/net/punycode.d

```diff
@@ -0,0 +1,232 @@
+/**
+	Encodes and decodes punycode strings, per
+	$(LINK2 https://www.ietf.org/rfc/rfc3492.txt, RFC3492).
+	Punycode is primarily used for converting URIs or IRIs containing non-ASCII
+	characters into URIs or IRIs that are entirely ASCII, and vice versa.
+	This punycode codec is based upon the original implementation found in
+	$(LINK2 https://www.ietf.org/rfc/rfc3492.txt, RFC3492).
+
+	Authors: Shotaro Yamada (Sinkuu)
+	Date: February 27, 2017
+	License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+	Standards: $(LINK2 https://www.ietf.org/rfc/rfc3492.txt, RFC3492)
+	Source: $(PHOBOSSRC std/net/_punycode.d)
+	Version: 1.0.1
+*/
+module punycode;
+
+private import std.ascii : isASCII, isUpper, isLower, isDigit;
```

Comment:
* `private` is the default for imports, no need to specify it
* please use whole module imports until bug 314/313 are both fully deprecated